neo/Core/Witness.py

```diff
@@ -12,37 +7,67 @@ class Witness(SerializableMixin):
     InvocationScript = None
     VerificationScript = None
 
-    def __init__(self, invocation_script=None, verification_script=None):
+    def __init__(self, invocation_script=bytearray(), verification_script=bytearray()):
+        """
+        Create an instance.
+
+        Args:
+            invocation_script (bytearray): the invocation script
+            verification_script (bytearray): the verification script
+
+        Throws:
+            Exception: if verification_script is a string
+        """
         try:
             self.InvocationScript = binascii.unhexlify(invocation_script)
         except Exception as e:
             self.InvocationScript = invocation_script
```

Comment:
What is a valid case for exception handling? If we submit a list `[0xaa,0xbb]` it catch it as expected, create the Witness instance but that will not be serializable

Reply:
The exception it catches is a `binascii` error when trying to `unhexlify` something that is already a raw byte string like `b'\x01\x02\xAF'` etc.  This could be made to catch a `binascii` error instead of a generic exception.

Reply:
How about we simplify it and just force the arguments to be of type `bytes`or `bytearray`?
e.g.
```
def __init__(self, invocation_script=bytearray(), verification_script=bytearray()):
    if not isinstance(invocation_script, (bytearray, bytes)):
        raise ValueError('Invalid invocation_script parameter type: {} is not of type bytearray or bytes'.format(type(invocation_script)))

    if not isinstance(verification_script, (bytearray, bytes)):
        raise ValueError('Invalid verification_script parameter type: {} is not of type bytearray or bytes '.format(type(verification_script)))

    self.InvocationScript = invocation_script
    self.VerificationScript = verification_script
```
All existing calls to `Witness` already adhere to these data types and `Deserialize()` also already returns `bytes`. This way we don't need to abuse any exception handling for type checking.

Reply:
both `bytes` and `bytearrays` can be in hexlified and unhexlfied format, so enforcing it to be one or the other won't help.

Reply:
Good point. I think instead of direct assignment we should still do the try/catch on `binascii.unhexlify` then, but also keep the instance testing. 

Why? Because a `string` is a valid argument type for `unhexlify` according to the documentation. However, it can throw the same `binascii.Error` exception when the string is invalid as an already raw bytes input. E.g.
1) `binascii.unhexlify(b'\x01\x02\xAF)` throws `binascii.Error('Odd-length string')
2) `binascii.unhexlify('aabbd')` throws the same error as 1.

The first value you still want to keep as it's actually valid but requires no unhexlifying, the second is really faulty. 

So excluding non-`bytes`/`bytearray` types would make it so that the only `binascii.Error` thrown by `unhexlify` is because of an already raw bytes value.

```
def __init__(self, invocation_script=bytearray(), verification_script=bytearray()):
        if not isinstance(invocation_script, (bytearray, bytes)):
            raise ValueError('Invalid invocation_script parameter type: {} is not of type bytearray or bytes'.format(type(invocation_script)))

        if not isinstance(verification_script, (bytearray, bytes)):
            raise ValueError('Invalid verification_script parameter type: {} is not of type bytearray or bytes '.format(type(verification_script)))

        try:
            self.InvocationScript = binascii.unhexlify(invocation_script)
        except binascii.Error:
            self.InvocationScript = invocation_script
            
        try:
            self.VerificationScript= binascii.unhexlify(verification_script)
        except binascii.Error:
            self.VerificationScript = verification_script
```

Reply:
@localhuman ping. I'd like to wrap this PR up so I can submit the next one that depends on this one :)

Reply:
That looks good! 👍